File: Backend/Business/Services/RoleService.cs
using Business.Interfaces;
using Business.Interfaces.Repository;
using Models.DTOs.Role;


namespace Business.Services
{
    public class RoleService : IRoleService
    {
        private IUnitOfWork _unitOfWork;
        public RoleService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }

        public async Task AddRole(CreateRoleDto role)
        {
            await _unitOfWork.Roles.AddRole(role);
            await _unitOfWork.SaveChangesAsync();
        }

        public async Task DeleteRole(int roleId)
        {
            await _unitOfWork.Roles.DeleteRole(roleId);
            await _unitOfWork.SaveChangesAsync();
        }

        public Task<IEnumerable<RoleDto>> GetAllRoles()
        {
            return _unitOfWork.Roles.GetRoles();
        }
        public Task<RoleDto> GetRoleByName(string roleName)
        {
            return _unitOfWork.Roles.GetRoleByName(roleName);
        }
        public Task<RoleDto> GetRoleById(int roleId)
        {
            return _unitOfWork.Roles.GetRoleById(roleId);
        }

        public async Task UpdateRole(UpdateRoleDto updateRoleDto)
        {
            await _unitOfWork.Roles.UpdateRole(updateRoleDto);
            await _unitOfWork.SaveChangesAsync();
        }
    }
}

File: Frontend/src/services/roleService.js
import api from "@/api/axios";

export const roleService = {

    async GetAllRoles() {
        const response = await api.get("/roles");

        return response.data;
    },

    async GetRoleById(id) {
        const response = await api.get(
            `/roles/${id}`
        );

        return response.data;
    },

    async CreateRole(roleData) {
        const response = await api.post(
            "/roles",
            roleData
        );

        return response.data;
    },

    async UpdateRole(id, roleData) {
        const response = await api.put(
            `/roles/${id}`,
            roleData
        );

        return response.data;
    },

    async DeleteRole(id) {
        const response = await api.delete(
            `/roles/${id}`
        );

        return response.data;
    }

};